tests/lib/external/snapd-testing-tools/utils/spread-filter

```diff
@@ -0,0 +1,223 @@
+        sorted_executions = sorted(executions, key=len, reverse=True)
+        for i in range(len(sorted_executions)):
+            discard_execution = False
+            long_execution = sorted_executions[i]
+            if self.verbose:
+                print("  checking repeated executions for {}".format(long_execution))
+
+            for j in range(i + 1, len(sorted_executions)):
+                short_execution = sorted_executions[j]
+                if self.verbose:
+                    print(
+                        "    comparing repeated executions for {}".format(
+                            short_execution
+                        )
+                    )
+
+                # if the shorter executions is a test, we continue
+                if not short_execution.endswith("/"):
+                    if long_execution == short_execution:
+                        discard_execution = True
+                        break
+                # if the shorter execution is a either suite or group of suites
+                else:
+                    # if the long executions starts with short execution means
+                    # the long execution is included in the short one and it can
+                    # be discarded
+                    if long_execution.startswith(short_execution):
+                        discard_execution = True
+                        break
+
+            # take decision after all shorter executions have been compared
+            if not discard_execution:
+                final_executions.append(long_execution)
+
+        return final_executions
+
+    # Retrieves the list of tests to execute
+    def get_executions(self, changes: list[str]) -> list[str]:
+        all_executions = []
+        for change in changes:
+            executions = self.rules.calc_executions(change)
+            # When the tests to run as tagged as SELF means the current test
+            # has to be executed
+            for execution in executions:
+                if execution == self.SELF:
+                    self_execution = self._calc_self(change)
+                    if self_execution != "":
+                        all_executions.append(self_execution)
+                # When the tests to run as tagged as NONE means that no tests
+                # have to be executed
+                elif execution == self.NONE:
+                    continue
+                # Otherwise, it is executed what is defined in to (literal)
+                else:
+                    all_executions.append(self._get_test_exec_dir(execution))
+
+        cleaned_executions = self._clean_executions(all_executions)
+        return self._get_execution_param(cleaned_executions)
+
+
+def _make_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(description="spread shellcheck helper")
+    parser.add_argument("-r", "--rules-file", required=True, help="Rules file")
+    parser.add_argument(
```

Comment:
Most parameters will always be required, so wondering if a notation closer to that of spread will be easier to use?
 - backend:system
 - search for config file based on naming convention (like spread.yaml)

The the end result could be e.g.
`spread-filter google:ubuntu -c source/lib1/lib1_test.go -c tests/main/test1/noexist.go`

Reply:
agree, I'll add a parameter called prefix which could be either backend:system or just the backend.